lib/mayaUsd/render/MaterialXGenOgsXml/GlslFragmentGenerator.cpp

```diff
@@ -385,8 +385,45 @@ ShaderPtr GlslFragmentGenerator::generate(
                 true);
         }
 
-        // Add all function calls
+        // Add all function calls (varies greatly by MaterialX version)
+#if MX_COMBINED_VERSION < 13803
         emitFunctionCalls(graph, context, pixelStage);
+#else
+        // Surface shaders need special handling.
+        if (graph.hasClassification(
+                ShaderNode::Classification::SHADER | ShaderNode::Classification::SURFACE)) {
+            // Emit all texturing nodes. These are inputs to any
+            // closure/shader nodes and need to be emitted first.
+            emitFunctionCalls(graph, context, pixelStage, ShaderNode::Classification::TEXTURE);
+
+#if MX_COMBINED_VERSION < 13805
```

Comment:
Having the nested `#if` and `#else` block makes it a bit difficult to figure out, what code will be run for what version. Maybe it would be more readable to have completely separate blocks for each version:
```C++
#if MX_COMBINED_VERSION < 13803
// v1.38.2 code
// ...

#elif MX_COMBINED_VERSION < 13805
// 1.38.3 code
// ...

#else
// 1.38.5 code
// ...

#endif
```

This would introduce some duplicated code, but I don't think it's a big problem because the older versions will probably remain unchanged. Feel free to leave it as is if you think it is worth it to avoid the code duplication.

Reply:
Agreed. I reformatted the code by blocks, and put the most current at the top to improve readability.

Reply:
Thanks!